api/app.py

```diff
@@ -19,58 +19,170 @@
+    'Pie': '#FFD700',  # gold
+    'Italian': '#FF0000',  # red
+    'No-bake': '#ADD8E6'  # lightblue
+}
+
+
+def get_categories_helper(recipe_id, categories):
+    categories_id = []
+    for recipe_cat in db.session.query(recipe_category).all():
+        if recipe_cat.recipe_id == recipe_id:
+            categories_id.append(recipe_cat.category_id)
+    for _id in categories_id:
+        for category in db.session.query(Category).all():
+            if category.id == _id:
+                categories.append(category.name)
+
+    return categories
+
+
+@app.route('/api/recipes', methods=['GET'])
 def get_recipes():
     recipes = []
     for recipe in db.session.query(Recipe).all():
-        recipes.append(recipe.as_dict())
-    return jsonify(recipes)
+        dict_recipe = recipe.as_dict()
+        categories = []
+
+        dict_recipe["ingredients"] = [ingredient.as_dict() for ingredient in recipe.ingredients]
+
+        categories = get_categories_helper(dict_recipe["id"], categories)
+        dict_recipe["categories"] = categories
 
-@app.route("/api/recipes/<int:recipe_id>" , methods = ['GET'])
+        recipes.append(dict_recipe)
+
+    return jsonify(recipes), 200
+
+
+@app.route("/api/recipes/<int:recipe_id>", methods=['GET'])
 def get_recipe(recipe_id):
     for recipe in db.session.query(Recipe).all():
         if recipe.as_dict().get("id") == recipe_id:
-            return jsonify(recipe.as_dict())
-    return jsonify({"error": "Recipe not found"}) , 404
+            dict_recipe = recipe.as_dict()
+
+            dict_recipe["ingredients"] = [ingredient.as_dict() for ingredient in recipe.ingredients]
+
+            categories = []
+            categories = get_categories_helper(dict_recipe["id"], categories)
+
+            dict_recipe["categories"] = categories
+
+            return jsonify(dict_recipe)
 
-#tema validare
-@app.route("/api/recipes" , methods = ["POST"])
+    return jsonify({"error": "Not found"}), 404
+
+
+@app.route("/api/recipes", methods=["POST"])
 def create_recipe():
-    new_recipe = {
-        "id": len(recipes) + 1,
-        "name": request.json.get("name"),
-        "duration": request.json.get("duration") ,
-        "pictures": request.json.get("pictures") ,
-        "instructions": request.json.get("instructions") ,
-        "ingredients": request.json.get("ingredients") ,
-        "categories": request.json.get("categories") ,
-    }
-    recipes.append(new_recipe)
-    return jsonify(new_recipe), 201
-
-@app.route("/api/recipes/<int:recipe_id>" , methods = ["DELETE"])
+    data = request.get_json()
+    fields = ["name", "duration", "instructions", "pictures", "categories"]
+    for field in fields:
+        if data.get(field) is None:
+            return jsonify({"error": f"Field {field} not found."}), 404
+
+    category_objs = []
+    for category_name in data.get("categories"):
+        category = Category.query.filter_by(name=category_name).first()
+        if not category:
```

Comment:
We don't need to create a `category` that doesn't exist here; we'll use the `create_category()` endpoint. Here you can return an error message if the category does not exist, for example:
```python
return jsonify({'error': f'category {cat["name"]} not found'}), 400
``` 
Same for `edit_recipe()`.